backpack/utils/conv.py

```diff
@@ -49,23 +49,37 @@ def separate_channels_and_pixels(module, tensor):
     return eingroup("v,n,c,h,w->v,n,c,hw", tensor)
 
 
-def extract_weight_diagonal(module, input, grad_output):
+def extract_weight_diagonal(module, input, grad_output, N):
     """
     input must be the unfolded input to the convolution (see unfold_func)
     and grad_output the backpropagated gradient
     """
-    grad_output_viewed = separate_channels_and_pixels(module, grad_output)
```

Comment:
Is this `separate_channels_and_pixels`function used somewhere else? If so, why not just generalize it to be able to handle `N=1,2,3`? If not, then deprecate that function

Reply:
Umm, we can generalize it. It's used in two places but the readability of the code will reduce

Reply:
Wouldn't it also be used by transpose convolution>

Reply:
yes!

Reply:
okay when I change this, `test_kfac_shape[Conv2d-ReLU-regression-cpu]` this fails

Reply:
Then, let's postpone this refactoring step until we have more extensive tests on KFAC and actually start extending it.